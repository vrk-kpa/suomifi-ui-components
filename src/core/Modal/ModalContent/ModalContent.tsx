import React, { Component, ReactNode } from 'react';
import { styled } from 'styled-components';
import classnames from 'classnames';
import { HtmlDiv, HtmlDivProps } from '../../../reset';
import { ModalConsumer, ModalVariant, baseClassName } from '../Modal/Modal';
import { baseStyles } from './ModalContent.baseStyles';
import { SuomifiThemeConsumer, SuomifiThemeProp } from '../../theme';

export interface ModalContentProps
  extends Omit<HtmlDivProps, 'children' | 'className'> {
  /** CSS class for custom styles */
  className?: string;
  /** Modal content */
  children: ReactNode;
}

interface InternalModalContentProps
  extends ModalContentProps,
    SuomifiThemeProp {
  id?: string;
  modalVariant: ModalVariant;
  scrollable: boolean;
}

const contentClassName = `${baseClassName}_content`;
const modalContentClassNames = {
  smallScreen: `${contentClassName}--small-screen`,
  noScroll: `${contentClassName}--no-scroll`,
  heading: `${contentClassName}_heading`,
};

class BaseModalContent extends Component<InternalModalContentProps> {
  render() {
    const {
      className,
      theme,
      title,
      children,
      scrollable,
      modalVariant,
      ...passProps
    } = this.props;

    return (
      <HtmlDiv
        className={classnames(contentClassName, className, {
          [modalContentClassNames.noScroll]: scrollable === false,
          [modalContentClassNames.smallScreen]: modalVariant === 'smallScreen',
        })}
        {...passProps}
      >
        {children}
      </HtmlDiv>
    );
  }
}

const StyledModalContent = styled(BaseModalContent)`
  ${({ theme }) => baseStyles(theme)}
`;

const ModalContent = (props: ModalContentProps) => (
  <SuomifiThemeConsumer>
    {({ suomifiTheme }) => (
      <ModalConsumer>
        {({ variant, scrollable }) => (
          <StyledModalContent
            theme={suomifiTheme}
            modalVariant={variant}
            scrollable={scrollable}
            {...props}
          />
        )}
      </ModalConsumer>
    )}
  </SuomifiThemeConsumer>
);

ModalContent.displayName = 'ModalContent';
export { ModalContent };
