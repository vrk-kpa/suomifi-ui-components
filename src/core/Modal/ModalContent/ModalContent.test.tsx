import React, { act } from 'react';
import { render } from '@testing-library/react';
import { ModalContent, ModalContentProps } from './ModalContent';
import { ModalTitle, Modal } from '../';
import { createRoot, Root } from 'react-dom/client';

let appRoot: HTMLDivElement | null = null;
let root: Root | null = null;

beforeEach(() => {
  appRoot = document.createElement('div');
  appRoot.setAttribute('id', 'root');
  document.body.appendChild(appRoot);
  root = createRoot(appRoot);
});

afterEach(() => {
  if (root) {
    act(() => {
      root?.unmount();
    });
    root = null;
  }
  if (appRoot) {
    appRoot.remove();
    appRoot = null;
  }
});

describe('Basic ModalContent', () => {
  const text = 'Modal Content';

  const BasicModal = (props?: Partial<ModalContentProps>) => (
    <Modal appElementId="root" visible={true}>
      <ModalContent {...props} data-testid="modal-content-id">
        <ModalTitle>Test modal</ModalTitle>
        <p>{text}</p>
        <button>x</button>
      </ModalContent>
    </Modal>
  );

  it('should have given title', () => {
    const { getByText } = render(BasicModal());
    expect(getByText('Test modal')).toBeTruthy();
  });

  it('should have given children', () => {
    const { getByText } = render(BasicModal());
    expect(getByText(text)).toBeTruthy();
  });

  it('should have given className', () => {
    const { getByTestId } = render(
      BasicModal({ className: 'test-class-name' }),
    );
    expect(getByTestId('modal-content-id')).toHaveClass('test-class-name');
  });

  it('should match snapshot', () => {
    const { baseElement } = render(BasicModal());
    expect(baseElement).toMatchSnapshot();
  });
});

describe('ModalContent variant', () => {
  const SmallScreenModal = (
    <Modal appElementId="root" visible={true} variant="smallScreen">
      <ModalContent data-testid="modal-content-id">
        <ModalTitle>Test modal</ModalTitle>
        <p>Modal Content</p>
        <button>x</button>
      </ModalContent>
    </Modal>
  );

  it('smallScreen should have correct classname', () => {
    const { getByTestId } = render(SmallScreenModal);
    expect(getByTestId('modal-content-id')).toHaveClass(
      'fi-modal_content--small-screen',
    );
  });
});
