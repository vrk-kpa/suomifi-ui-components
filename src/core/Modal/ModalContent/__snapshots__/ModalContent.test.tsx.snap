// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic ModalContent should match snapshot 1`] = `
.c1 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1 ::before,
.c1 ::after {
  box-sizing: border-box;
}

.c5 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c5 ::before,
.c5 ::after {
  box-sizing: border-box;
}

.c3 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c3 ::before,
.c3 ::after {
  box-sizing: border-box;
}

.c0.fi-modal_base {
  color: hsl(0, 0%, 13%);
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
}

.c0.fi-modal_base .fi-modal_overlay {
  background-color: rgba(33,33,33,0.5);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.c0 .fi-modal {
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  border-top: 4px solid hsl(212, 63%, 45%);
  overflow: hidden;
  max-height: calc(100% - 50px);
  min-height: 230px;
  width: 800px;
  max-width: 100%;
  flex: 0 1 auto;
  display: flex;
  flex-direction: column;
  align-items: stretch;
}

.c0 .fi-modal:focus {
  outline: none;
}

.c0 .fi-modal:focus-visible {
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196, 77%, 44%);
  outline: 0;
}

.c0 .fi-modal--no-scroll {
  width: 540px;
  min-height: 190px;
}

.c0 .fi-modal--small-screen {
  border-radius: 0;
  width: 100%;
  height: 100%;
  max-height: 100%;
}

.c2 {
  color: hsl(0, 0%, 13%);
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c2.fi-modal_content {
  flex: 1 1 auto;
  max-height: 100%;
  padding: 24px 30px 50px 30px;
  overflow-y: auto;
  scroll-padding-bottom: 75px;
  scroll-padding-top: 75px;
}

.c2.fi-modal_content--no-scroll {
  overflow: hidden;
  padding: 34px 30px 10px 30px;
}

.c2.fi-modal_content--small-screen {
  padding-top: 20px;
  padding-left: 15px;
  padding-bottom: 50px;
  padding-right: 15px;
}

.c6 {
  color: hsl(0, 0%, 13%);
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  color: hsl(0, 0%, 13%);
}

.c6.fi-heading--h1hero {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 40px;
  line-height: 48px;
  font-weight: 600;
}

.c6.fi-heading--h1 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 40px;
  line-height: 48px;
  font-weight: 300;
}

.c6.fi-heading--h2 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 28px;
  line-height: 34px;
  font-weight: 600;
}

.c6.fi-heading--h3 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 22px;
  line-height: 28px;
  font-weight: 600;
}

.c6.fi-heading--h4 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
}

.c6.fi-heading--h5 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c6.fi-heading--h6 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c6.fi-heading--small-screen {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 300;
}

.c6.fi-heading--small-screen.fi-heading--h1hero {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 600;
}

.c6.fi-heading--small-screen.fi-heading--h1 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 300;
}

.c6.fi-heading--small-screen.fi-heading--h2 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 26px;
  line-height: 32px;
  font-weight: 600;
}

.c6.fi-heading--small-screen.fi-heading--h3 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 20px;
  line-height: 26px;
  font-weight: 600;
}

.c6.fi-heading--small-screen.fi-heading--h4 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c6.fi-heading--small-screen.fi-heading--h5 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c6.fi-heading--small-screen.fi-heading--h6 {
  letter-spacing: 0;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c4 {
  max-width: 100%;
}

.c4.fi-modal_title {
  display: block;
  margin-bottom: 20px;
}

.c4.fi-modal_title .fi-modal_title_focus-wrapper {
  position: relative;
  display: inline-block;
  max-width: 100%;
}

.c4.fi-modal_title .fi-modal_title_focus-wrapper:focus {
  outline: 0;
}

.c4.fi-modal_title .fi-modal_title_focus-wrapper:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0, 0%, 100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196, 77%, 44%);
  z-index: 9999;
}

.c4.fi-modal_title--small-screen {
  margin-bottom: 20px;
}

<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="c0 fi-modal_base"
  >
    <div
      class="ReactModal__Overlay fi-modal_overlay"
    >
      <div
        aria-modal="false"
        class="ReactModal__Content fi-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="c1 fi-modal_content c2"
          data-testid="modal-content-id"
        >
          <span
            class="c3 c4 fi-modal_title"
          >
            <h2
              class="c5 fi-heading c6 fi-modal_title_focus-wrapper fi-heading--h3"
              tabindex="0"
            >
              Test modal
            </h2>
          </span>
          <p>
            Modal Content
          </p>
          <button>
            x
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
